bundles/http/Citrix/SAP-Ariba/Ariba_v2.js

```diff
@@ -235,14 +238,24 @@ async function fetchChangesGetDocumentsAndGetLastId(sync, documentsLists) {
                     noOfPages = calculateNumberOfPages(+responseData.headers.get('X-Total-Count'), limitChanges)
                     console.log('Total number of changes ' + responseData.headers.get('X-Total-Count'))
                     console.log('Total number of pages for changes endpoint ' + noOfPages)
-                    console.log('First Change ID ' + json[0].changeSequenceId)
+                    console.log('First item with change ID ',json[0].changeSequenceId)
                 }
-
-                if (documentsLists.newChangeId < +json[json.length - 1]
-                    .changeSequenceId) { //getting the  lastChangeSequenceId for incremental sync from page
-                    documentsLists.newChangeId = +json[json.length - 1].changeSequenceId
+                              
+                try {
```

Comment:
this code block is weird, I know that JSON is used instead of object but now it looks like JSON is also synonym for an array? what is `+(...)`? it's unnecessary operator. error message is missing som clue from the caught error

Reply:
json[0].changeSequenceId Apiba api sometimes return it as a string and sometimes as number or simetimes is missing at all event it's PK for the endpoint, but actually is a number so + just foce number. Error message appears only when api will not send the value at all...